src/services/utilities.ts

```diff
@@ -562,6 +562,15 @@ namespace ts {
         return kind === SyntaxKind.SingleLineCommentTrivia || kind === SyntaxKind.MultiLineCommentTrivia;
     }
 
+    export function isStringOrRegularExpressionOrTemplateLiteral(kind: SyntaxKind): boolean {

```

Comment:
can you also replace a bunch of checks [here] (https://github.com/Microsoft/TypeScript/blob/master/src/services/formatting/smartIndenter.ts#L22) with call to this function?